xdr/signer_key.go

```diff
@@ -40,6 +41,18 @@ func (skey *SignerKey) GetAddress() (string, error) {
 		vb = strkey.VersionByteHashTx
 		key := skey.MustPreAuthTx()
 		copy(raw, key[:])
+	case SignerKeyTypeSignerKeyTypeEd25519SignedPayload:
+		sp := skey.MustEd25519SignedPayload()
+		publicKey, err := strkey.Encode(strkey.VersionByteAccountID, sp.Ed25519[:])
+		if err != nil {
+			return "", errors.Wrap(err, "failed to decode payload signer")
+		}
+		spWrapper, err := strkey.MakeSignedPayload(publicKey, sp.Payload)
+		if err != nil {
+			return "", errors.Wrap(err, "failed to create signed payload")
+		}
+		return spWrapper.Encode()
```

Comment:
💭 Something we might want to come back to and improve is these lines of code encode a G strkey, then decode it, then re-encode it in a P strkey. Strkey's functions have been optimized, but they have to allocate unfortunately. Let's not worry about this now, but I'm thinking a `strkey.NewEd25519SignedPayload` that accepts a `ed25519 byte[32]`. I can add that when testing Starlight if I find Starlights benchmarks are adversely affected.

Technically you could also just call the following too which wouldn't do so much encoding/decoding.
```suggestion
		raw, err = sp.MarshalBinary()
		if err != nil {
			return "", err
		}
```

Reply:
Yeah I wasn't happy about the back-and-forth there, but wow it's so much simpler to just marshal it directly.. nice, thanks!